src/server.tsx

```diff
@@ -44,34 +45,45 @@ interface Response {
     css: string;
     js: string;
     meta: EpicTestTracking;
+    debug?: Debug;
 }
 
 const [, fetchConfiguredEpicTestsCached] = cacheAsync(fetchConfiguredEpicTests, 60);
 
-const asResponse = (component: JsComponent, meta: EpicTestTracking): Response => {
+const asResponse = (component: JsComponent, meta: EpicTestTracking, debug?: Debug): Response => {
     const { el, js } = component;
     const { html, css } = extractCritical(renderToStaticMarkup(el));
-    return { html, css, js, meta };
+    return { html, css, js, meta, debug };
 };
 
 const buildEpic = async (
     pageTracking: EpicPageTracking,
     targeting: EpicTargeting,
+    params: Params,
 ): Promise<Response | null> => {
     const configuredTests = await fetchConfiguredEpicTestsCached();
     const hardcodedTests = await getAllHardcodedTests();
     const tests = [...configuredTests.tests, ...hardcodedTests];
-    const result = findTestAndVariant(tests, targeting);
+
+    let result: Result;
+
+    if (params.force) {
+        const test = tests.find(test => test.name === params.force?.testName);
+        const variant = test?.variants.find(v => v.name === params.force?.variantName);
+        result = test && variant ? { test, variant } : {};
+    } else {
+        result = findTestAndVariant(tests, targeting, params.debug);
+    }
 
     logTargeting(
         `Renders Epic ${result ? 'true' : 'false'} for targeting: ${JSON.stringify(targeting)}`,
     );
 
-    if (!result) {
+    if (!result.test || !result.variant) {
```

Comment:
I wonder if having to check both `test` and `variant` here is a sign that the returned result doesn't have quite the right structure. Perhaps the debug could be separated out a little more? What if we returned a tuple from `findTestAndVariant` rather than having `debug` in the same object? Or an object structured like: `{ result: { test, variant }, debug }`.

Reply:
Yeah, I do agree that this is not great for the reason you say. Will change.